browser/resources/settings/email_aliases_page/content/email_aliases_list.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import ButtonMenu from '@brave/leo/react/buttonMenu'
+import Col from './styles/Col'
+import Description from './styles/Description'
+import formatMessage from '$web-common/formatMessage'
+import Icon from '@brave/leo/react/icon'
+import Row from './styles/Row'
+import styled from 'styled-components'
+import Tooltip from '@brave/leo/react/tooltip'
+
+const AliasItemRow = styled(Row)`
+  font: ${font.default.regular};
+  padding: ${spacing.l} ${spacing['2Xl']};
+  border-top: ${color.divider.subtle} 1px solid;
+  justify-content: space-between;
+`
+
+const AliasAnnotation = styled.div`
+  font: ${font.small.regular};
+  color: ${color.text.secondary};
+`
+
+const AliasListIntro = styled(Row)`
+  justify-content: space-between;
+  padding: ${spacing.l} ${spacing['2Xl']};
+  & leo-button {
+    flex-grow: 0;
+    font: ${font.components.buttonSmall};
+  }
+`
+
+const EmailContainer = styled.div`
+  cursor: pointer;
+  font: ${font.default.regular};
+`
+
+const AliasControls = styled(Row)`
+  user-select: none;
+`
+
+const ButtonUnpadded = styled(Button)`
+  --leo-button-padding: 0;
+  & leo-icon {
+    color: ${color.icon.default};
+  }
+`
+
+const DivWithTopDivider = styled.div`
+  border-top: ${color.divider.subtle} 2px solid;
+`
+
+const AliasMenuItem = ({ onClick, iconName, text }:
+  { onClick: EventListener, iconName: string, text: string }) =>
+  <leo-menu-item
+    onClick={onClick}>
+    <Row>
+      <Icon name={iconName} />
+      <span>{text}</span>
+    </Row>
+  </leo-menu-item>
+
+const CopyToast = ({ text, tabIndex, children }: { text: string, tabIndex?: number, children: React.ReactNode }) => {
+  const [copied, setCopied] = React.useState<boolean>(false)
+  const copy = () => {
+    navigator.clipboard.writeText(text)
+    setCopied(true)
+    setTimeout(() => setCopied(false), 1000)
+  }
+  return <div
+    tabIndex={tabIndex}
+    onKeyDown={onEnterKeyForDiv(copy)}
+    onClick={copy}>
+    <Tooltip text={copied ? getLocale('emailAliasesCopiedToClipboard') : ''} mode="mini" visible={copied}>
+      {children}
+    </Tooltip>
+  </div>
+}
+
+const AliasItem = ({ alias, onEdit, onDelete }: { alias: Alias, onEdit: () => void, onDelete: () => void }) =>
+  <AliasItemRow>
+    <Col>
+      <CopyToast text={alias.email}>
+        <EmailContainer title={getLocale('emailAliasesClickToCopyAlias')}>
+            {alias.email}
+          </EmailContainer>
+        </CopyToast>
+        {(alias.note || alias.domains) &&
+          <AliasAnnotation>
+            {alias.note}
+            {alias.domains && alias.note && <span>. </span>}
+            {alias.domains && formatMessage(getLocale('emailAliasesUsedBy'),
+              { placeholders: { $1: alias.domains?.join(", ") } })}
+          </AliasAnnotation>}
+        </Col>
+      <AliasControls>
+        <CopyToast text={alias.email} tabIndex={0}>
+          <ButtonUnpadded
```

Comment:
*note:* you might want to specify `fab` on these buttons (to make them rounded)

Reply:
done